dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/CheckCanAccessStepAttribute.cs

```diff
@@ -21,9 +21,16 @@ public void OnResourceExecuting(ResourceExecutingContext context)
     {
         var journey = context.HttpContext.RequestServices.GetRequiredService<SignInJourney>();
 
-        if (!journey.CanAccessStep(StepName))
+        try
         {
-            context.Result = new RedirectResult(journey.GetLastAccessibleStepUrl());
+            if (!journey.CanAccessStep(StepName))
+            {
+                context.Result = new RedirectResult(journey.GetLastAccessibleStepUrl());
+            }
+        }
+        catch (ArgumentOutOfRangeException)
+        {
+            context.Result = new BadRequestResult();
```

Comment:
This causes pages to respond with 400 rather than 500 which is possibly neater. It also helps pass tests which would otherwise have to Assert the exception was thrown

Reply:
Is it `GetLastAccessibleStepUrl` that's throwing `ArgumentOutOfRangeException`? Could change that to `bool TryGetLastAccessibleStepUrl(out string? url)` to avoid `try/catch`.

Reply:
Unfortunately the exception is also thrown from `CanAccessStep` so needs to be caught there too